cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -190,39 +197,84 @@ VAR_LEN_TYPES(IS_NOT_NULL, isnotnull)
 
 #undef IS_NOT_NULL
 
+/*
+ We follow Oracle semantics for offset:
+ - If position is positive, then the first glyph in the substring is determined by
+ counting that many glyphs forward from the beginning of the input. (i.e., for position ==
+ 1 the first glyph in the substring will be identical to the first glyph in the output)
```

Comment:
Sorry, this typo was my fault
```suggestion
 1 the first glyph in the substring will be identical to the first glyph in the input)
```